packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -0,0 +1,60 @@
+import { EthExecutionAPI, PromiEvent, ReceiptInfo } from 'web3-common';
+import { Web3Context } from 'web3-core';
+import { numberToHex } from 'web3-utils';
+
+import {
+	TransactionMissingReceiptOrBlockHashError,
+	TransactionReceiptMissingBlockNumberError,
+} from '../errors';
+import { ReceiptInfoFormatted, SendSignedTransactionEvents, SendTransactionEvents } from '../types';
+import { getBlockByNumber } from '../rpc_methods';
+
+export function watchTransactionForConfirmations<
+	PromiEventEventType extends SendTransactionEvents | SendSignedTransactionEvents,
+>(
+	web3Context: Web3Context<EthExecutionAPI>,
+	transactionPromiEvent: PromiEvent<ReceiptInfoFormatted, PromiEventEventType>,
+	transactionReceipt: ReceiptInfo,
+) {
+	if (
+		transactionReceipt === undefined ||
+		transactionReceipt === null ||
+		transactionReceipt.blockHash === undefined ||
+		transactionReceipt.blockHash === null
+	)
+		throw new TransactionMissingReceiptOrBlockHashError({
+			receipt: transactionReceipt,
```

Comment:
if transactionReceipt is `undefined` or `null`, will be difficult to debug this error so good to pass [transactionHash](https://github.com/ChainSafe/web3.js/blob/a09482b0eeab653f56a2147fd236a643688e68a2/packages/web3-eth/src/rpc_method_wrappers.ts#L377) to `watchTransactionForConfirmations` [here](https://github.com/ChainSafe/web3.js/blob/a09482b0eeab653f56a2147fd236a643688e68a2/packages/web3-eth/src/rpc_method_wrappers.ts#L401) and show that in detail of error.

Reply:
[Updated](https://github.com/ChainSafe/web3.js/pull/4875/commits/175c6d66debbd177040de8d462738b1784e8f982)